src/dtx/dtx_resync.c

```diff
@@ -148,66 +190,65 @@ dtx_status_handle(struct dtx_resync_args *dra)
 
 		rc = dtx_check(dra->po_uuid, cont->sc_uuid,
 			       &dre->dre_dte, layout);
-		if (rc != DTX_ST_COMMITTED && rc != DTX_ST_PREPARED) {
+
+		/* The DTX has been committed (or) ready to be committed on
+		 * some remote replica(s), let's commit the it globally.
+		 */
+		if (rc == DTX_ST_COMMITTED || rc == DTX_ST_PREPARED)
+			goto commit;
```

Comment:
We will check local status before finally adding it into the CoS. If local DTX is not committed yield, it will be added (if it has not been added yet).